homeassistant/components/azuredns/__init__.py

```diff
@@ -0,0 +1,142 @@
+CONF_TENANT = 'tenant'
+CONF_SUBSCRIPTIONID = 'subscriptionid'
+CONF_RESOURCEGROUPNAME = 'resourcegroupname'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DOMAIN): cv.string,
+        vol.Optional(CONF_HOST, default='@'): cv.string,
+        vol.Required(CONF_TENANT): cv.string,
+        vol.Required(CONF_CLIENTID): cv.string,
+        vol.Required(CONF_CLIENTSECRET): cv.string,
+        vol.Required(CONF_SUBSCRIPTIONID): cv.string,
+        vol.Required(CONF_RESOURCEGROUPNAME): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Initialize the Azure DNS component."""
+    domain = config[DOMAIN][CONF_DOMAIN]
+    host = config[DOMAIN][CONF_HOST]
+    resource = (RESOURCE)
+    tenant = config[DOMAIN][CONF_TENANT]
+    clientid = config[DOMAIN][CONF_CLIENTID]
+    clientsecret = config[DOMAIN][CONF_CLIENTSECRET]
+    authority_url = (AUTHORITYHOSTURL + '/' + config[DOMAIN][CONF_TENANT])
+    apiurl = ('https://management.azure.com/subscriptions/'
+              + config[DOMAIN][CONF_SUBSCRIPTIONID]
+              + '/resourceGroups/'
+              + config[DOMAIN][CONF_RESOURCEGROUPNAME]
+              + 'providers/Microsoft.Network/dnsZones/'
+              + config[DOMAIN][CONF_DOMAIN]
+              + '/A/'
+              + config[DOMAIN][CONF_HOST]
+              + '?api-version=2018-03-01-preview')
+
+    session = async_get_clientsession(hass)
+
+    result = await _update_azuredns(session, domain, host, resource, tenant,
+                                    clientid, clientsecret,
+                                    authority_url, apiurl)
+
+    if not result:
+        return False
+
+    async def update_domain_interval(now):
+        """Update the Azure DNS entry."""
+        await _update_azuredns(session, domain, host, resource, tenant,
+                               clientid, clientsecret, authority_url, apiurl)
+
+    async_track_time_interval(hass, update_domain_interval, INTERVAL)
+    return result
+
+
+async def _update_azuredns(session, domain, host, resource, tenant,
+                           clientid, clientsecret, authority_url, apiurl):
+    import adal
+    import json
+
+    params = {
+        'domain': domain,
+        'host': host,
+        'resource': resource,
+        'tenant': tenant,
+        'clientid': clientid,
+        'clientsecret': clientsecret,
+        'apiurl': apiurl,
+    }
+
+    """Requesting the Azure AD App Token by using ADAL."""
+    context = adal.AuthenticationContext(
+        authority_url, validate_authority=['tenant'] != 'adfs',
+    )
+
+    token = context.acquire_token_with_client_credentials(
+        params['resource'],
+        params['clientid'],
+        params['clientsecret'])
+
+    access_token = token.get('accessToken')
+
+    _LOGGER.debug("Azure AD App Token is:\n" + json.dumps(token, indent=2))
+
+    """Update the DNS entry by using the Azure REST API."""
+    import requests
```

Comment:
I think that we should go with the `aiohttp_client` and not using `requests` additionally.

Reply:
Thanks, it's resolved in b23d27d49474726e9cde4040ad6ef6e8b209fdd3.